src/lib/slice.zig

```diff
@@ -452,13 +525,15 @@ pub const Slice = struct {
     }
 
     /// Reads the parent branch from git notes for this slice
-    /// TODO: This logic is not robust since it only checks the latest commit in the branch for the note.
-    /// We might need to check all notes in the branch since the target of the feature and only accept
-    /// the latest `slice-parent:` to be source of truth.
+    /// Uses slice-specific namespace to avoid conflicts between slices
     pub fn getParentFromNotes(self: *const Slice, alloc: Allocator) !?[]const u8 {
         const commit_oid = self.ref.target() orelse return null;
 
-        const note = LibGit.GitNote.read(self.repo, commit_oid) catch |err| {
+        // Get the slice-specific notes namespace
+        const notes_namespace = try self.sliceBranchToNotesNamespace(alloc);
+        defer alloc.free(notes_namespace);
+
+        const note = LibGit.GitNote.read(self.repo, commit_oid, notes_namespace) catch |err| {
```

Comment:
instead of reading note from HEAD of the slice branch. `commit_oid` is here points the latest commit in the slice. Instead searching through the notes in the namespace of slice and  taking the last one as source of truth would be better option here to determine the parent of a slice.